pkg/server/connect_interceptors/logger.go

```diff
@@ -51,29 +51,30 @@ func UnaryConnectLoggerInterceptor(logger *zap.Logger, opts *LoggerOptions) conn
 			resp, err := next(ctx, req)
 			duration := time.Since(startTime)
 
-			// Get response code from error or OK if no error
-			code := connect.Code(0).String()
-			if err != nil {
-				if connectErr, ok := err.(*connect.Error); ok {
-					code = connectErr.Code().String()
-				}
+			code := connect.Code(0)
+			if connectErr, ok := err.(*connect.Error); ok {
+				code = connectErr.Code()
 			}
 
 			fields := []zap.Field{
 				zap.String("system", "connect_rpc"),
 				zap.Time("start_time", startTime),
 				zap.String("method", req.Spec().Procedure),
 				zap.Int64("time_ms", duration.Milliseconds()),
-				zap.String("code", code),
+				zap.String("code", code.String()),
 				zap.String("request_id", req.Header().Get(consts.RequestIDHeader)),
 				zap.Error(err),
 			}
-			if err == nil {
+			if err == nil && ctx.Err() == nil {
```

Comment:
since we are not handling context related errors in this interceptor, do we need this change?